src/migrations/merge.rs

```diff
@@ -0,0 +1,147 @@
+use indexmap::IndexMap;
+use fs_err as fs;
+
+use crate::connect::Connection;
+use crate::migrations::{Context, migrate, migration};
+use crate::migrations::create::{MigrationKey, MigrationToText, write_migration};
+use crate::migrations::db_migration::{DBMigration, read_all};
+use crate::migrations::migration::MigrationFile;
+use crate::migrations::rebase::{fix_migration_ids};
+
+pub struct MergeMigrations {
+    pub base_migrations: IndexMap<String, MergeMigration>,
+    pub target_migrations: IndexMap<String, MergeMigration>
+}
+
+impl MergeMigrations {
+    fn flatten(&self) -> IndexMap<&String, &MergeMigration> {
+        let mut result = IndexMap::new();
+
+        for migration in self.base_migrations.iter() {
+            result.insert(migration.0, migration.1);
+        }
```

Comment:
```suggestion
        result.insert(self.base_migrations);
```
(I'm not sure this will work, but it would with normal `Map` or `Vec`)

Reply:
seems to work fine? why would it not normally work?

Reply:
No it doesn't seem to work with `IndexMap`